src/clr-addons/datagrid/enum-filter/enum-filter.component.ts

```diff
@@ -17,25 +22,26 @@ export class ClrEnumFilterComponent<T extends { [key: string]: any }>
 
   @Input('clrFilterValues')
   public set value(values: string[]) {
+    const converted = values.map(filtered => ({ value: filtered, displayValue: filtered }));
     if (this.possibleValues?.length) {
-      this.filteredValues = values.filter(filtered => this.possibleValues.includes(filtered));
-      this.clrFilterValuesChange.emit(this.filteredValues);
+      this.filteredValues = converted.filter(filtered => this.possibleValues.includes(filtered));
+      this.clrFilterValuesChange.emit(values);
```

Comment:
this should only contain values from "this.filteredValues", not all given as input parameters

Reply:
changed